src/gtkutils.c

```diff
@@ -35,50 +35,71 @@ GtkWidget *ut_window_new(gchar *title, gchar *wmname, gchar *wmclass,
 }
 
 
-GtkWidget *ut_check_button_new(gchar *label, gboolean active, GtkWidget *parent)
+GtkWidget * gtkutil_dialog_new (char * title,
+                                GtkWindow * parent,
+                                gboolean resizable,
+                                GtkWidget ** main_vbox) /* out */
 {
-   GtkWidget *button;
+   GtkWidget * dialog;
+   dialog = gtk_dialog_new ();
+   gtk_window_set_title (GTK_WINDOW (dialog), title);
+   gtk_container_set_border_width (GTK_CONTAINER (dialog), 4); /* padding */
+   if (parent) {
+      gtk_window_set_transient_for (GTK_WINDOW (dialog), GTK_WINDOW (parent));
+      gtk_window_set_modal (GTK_WINDOW (dialog), TRUE);
+      gtk_window_set_destroy_with_parent (GTK_WINDOW (dialog), TRUE);
+      gtk_window_set_position (GTK_WINDOW (dialog), GTK_WIN_POS_CENTER_ON_PARENT);
+      gtk_window_set_skip_pager_hint (GTK_WINDOW (dialog), TRUE);
+      gtk_window_set_skip_taskbar_hint (GTK_WINDOW (dialog), TRUE);
+   }
+   if (!resizable) {
+      // no need to call this if TRUE, unexpected behavior in GTK3 IIRC
+      gtk_window_set_resizable (GTK_WINDOW (dialog), FALSE);
+   }
 
-   button = gtk_check_button_new_with_label(label);
-   gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(button), active);
-   gtk_box_pack_start(GTK_BOX(parent), button, FALSE, FALSE, 0);
+   if (main_vbox) {
+      *main_vbox = gtk_dialog_get_content_area (GTK_DIALOG (dialog));
+      /* padding */
+      gtk_container_set_border_width (GTK_CONTAINER (*main_vbox), 4);
+   }
 
-   return button;
+   return (dialog);
```

Comment:
Hmm, isn't `return` a statement in C? Why extra parens if it's not a function call.